src/net_processing.cpp

```diff
@@ -3292,6 +3367,56 @@ void PeerManagerImpl::ProcessCompactBlockTxns(CNode& pfrom, Peer& peer, const Bl
     return;
 }
 
+void PeerManagerImpl::PushUnbroadcastTx(CNode& node, Peer& peer, const std::atomic<bool>& interrupt)
+{
+    Peer::TxRelay* tx_relay{peer.GetTxRelay()};
+    if (tx_relay == nullptr) {
+        LogPrintLevel(BCLog::SENSITIVE_RELAY, /* Continued */
+                      BCLog::Level::Debug,
+                      "Disconnecting: no transactions relay with this peer (connected in vain), peer=%d%s\n",
+                      node.GetId(),
+                      fLogIPs ? strprintf(", peeraddr=%s", node.addr.ToStringAddrPort()) : "");
+        node.fDisconnect = true;
+        m_connman.ScheduleSensitiveRelayConnections(1);
```

Comment:
> IIUC this line schedules a replacement sensitive connection because the one we were passed in has the wrong services and so it gets dumped.

Yes.

> I think this is the only line in this function that makes it apply strictly to only sensitive relays. So I wonder if that should be asserted at the top? Or could be renamed to `PushUnbroadcastTxToSensistiveRelay()`? At the very least maybe sensitive-only should be explained in the function description on L933 ?

Right. While this function will work if called for non-sensitive-relay peer (it will send the tx to the peer), this is not the intention because it will be bad for privacy and it is only called under `if (pfrom.IsSensitiveRelayConn()) {`.

Renamed as suggested and added `Assume()`;